Intersect.Client/Interface/Game/DescriptionWindows/ItemDescriptionWindow.cs

```diff
@@ -324,11 +324,17 @@ protected void SetupEquipmentInfo()
             }
 
             // Stats
-            if (mItemProperties?.StatModifiers != default)
+            for (var i = 0; i < (int)Stat.StatCount; i++)
             {
-                for (var i = 0; i < (int)Stat.StatCount; i++)
+                // Do we have item properties, if so this is a finished item. Otherwise does this item not have growing stats?
+                if (mItemProperties != default || mItem.StatGrowth == 0)
                 {
-                    var flatStat = mItem.StatsGiven[i] + mItemProperties.StatModifiers[i];
+                    var flatStat = mItem.StatsGiven[i];
+                    if (mItemProperties != default)
+                    {
+                        flatStat += mItemProperties.StatModifiers[i];
```

Comment:
Can you just save StatModifiers into a local variable before the for loop and check it ahead of time to make sure we can't NPE in the event of a race condition?

Reply:
To my knowledge mItemProperties is already a local variable is it not?

https://github.com/AscensionGameDev/Intersect-Engine/blob/34b1aa733d6b9de114a2d7fd27270293c5b13a68/Intersect.Client/Interface/Game/DescriptionWindows/ItemDescriptionWindow.cs#L40